### PR TITLE
Changed return values of SectionHeaderIterator().next() to bswapAllFields()

### PR DESCRIPTION
**Pull request by Artix-Fox**
SectionHeaderIterator().next() now uses bswapAllFields()

### CODE REVIEW

**Comment by g-w1** [comment]
Please run zig fmt on the files you changed.

**Comment by Artix-Fox** [comment]
wait, i thought that doom emacs did it by default. Sorry

File: lib/std/elf.zig [hunk 1]
```diff
@@ -479,18 +479,7 @@ pub fn SectionHeaderIterator(ParseSource: anytype) type {
...
-                    .sh_flags = @byteSwap(@TypeOf(shdr.sh_flags), shdr.sh_flags),
-                    .sh_addr = @byteSwap(@TypeOf(shdr.sh_addr), shdr.sh_addr),
-                    .sh_offset = @byteSwap(@TypeOf(shdr.sh_offset), shdr.sh_offset),
-                    .sh_size = @byteSwap(@TypeOf(shdr.sh_size), shdr.sh_size),
-                    .sh_link = @byteSwap(@TypeOf(shdr.sh_link), shdr.sh_link),
-                    .sh_info = @byteSwap(@TypeOf(shdr.sh_info), shdr.sh_info),
-                    .sh_addralign = @byteSwap(@TypeOf(shdr.sh_addralign), shdr.sh_addralign),
-                    .sh_entsize = @byteSwap(@TypeOf(shdr.sh_entsize), shdr.sh_entsize),
-                };
+                bswapAllFields(Elf64_Shdr, &shdr);
```
Comment (Vexu) [inline lib/std/elf.zig:482]: There should ve a return after this.

**Review by Vexu — CHANGES_REQUESTED** [review]

File: lib/std/elf.zig (hunk 1 above)
Comment (Artix-Fox) [reply lib/std/elf.zig]: changed, also sorry for the commit as i wasn't able to force push the changes.

File: lib/std/elf.zig [hunk 2]
```diff
@@ -479,18 +479,7 @@ pub fn SectionHeaderIterator(ParseSource: anytype) type {
...
-                    .sh_flags = @byteSwap(@TypeOf(shdr.sh_flags), shdr.sh_flags),
-                    .sh_addr = @byteSwap(@TypeOf(shdr.sh_addr), shdr.sh_addr),
-                    .sh_offset = @byteSwap(@TypeOf(shdr.sh_offset), shdr.sh_offset),
-                    .sh_size = @byteSwap(@TypeOf(shdr.sh_size), shdr.sh_size),
-                    .sh_link = @byteSwap(@TypeOf(shdr.sh_link), shdr.sh_link),
-                    .sh_info = @byteSwap(@TypeOf(shdr.sh_info), shdr.sh_info),
-                    .sh_addralign = @byteSwap(@TypeOf(shdr.sh_addralign), shdr.sh_addralign),
-                    .sh_entsize = @byteSwap(@TypeOf(shdr.sh_entsize), shdr.sh_entsize),
-                };
+                return bswapAllFields(Elf64_Shdr, &shdr);
```
Comment (Vexu) [inline lib/std/elf.zig:482]:
```suggestion
                bswapAllFields(Elf64_Shdr, &shdr);
                return shdr;
```
`bswapAllFields` returns void.

**Review by Vexu — CHANGES_REQUESTED** [review]

**Comment by Vexu** [comment]
Thanks! Std cleanups are always appreciated.

**Pull request merged**